@here/harp-mapview/lib/MapView.ts

```diff
@@ -2709,30 +2709,32 @@ export class MapView extends THREE.EventDispatcher {
         if (this.m_drawing) {
             return;
         }
-        ++this.m_frameNumber;
 
+        let currentFrameEvent: FrameStats | undefined;
         const stats = PerformanceStatistics.instance;
         const gatherStatistics: boolean = stats.enabled;
+        if (gatherStatistics) {
+            currentFrameEvent = stats.currentFrame;
+            // We store the last frame statistics b/c additional work (i.e. geometry creation)
+            // is done outside of the animation frame but still needs to be added to the
+            // `fullFrameTime`
+
+            // For convenience and easy readability
+            const fullFrameTime = currentFrameEvent.getValue("render.fullFrameTime");
+            currentFrameEvent.setValue("render.fps", 1000 / fullFrameTime!);
 
+            stats.storeAndClearFrameInfo();
+        }
+
+        ++this.m_frameNumber;
         const frameStartTime = time;
 
         RENDER_EVENT.time = time;
         this.dispatchEvent(RENDER_EVENT);
 
-        let currentFrameEvent: FrameStats | undefined;
-
         if (gatherStatistics) {
-            currentFrameEvent = stats.currentFrame;
+            currentFrameEvent = currentFrameEvent as FrameStats;
```

Comment:
Although I understand this convention is look little bit awkward :).

Reply:
Yeah, I was to lazy to add `!` everywhere